From a4d18f445ae195667f1fbeb9ec600f76383c9a28 Mon Sep 17 00:00:00 2001
From: Martin Braun <martin.braun@ettus.com>
Date: Mon, 27 Apr 2020 19:24:58 -0700
Subject: [PATCH 15/32] uhd: Change import order for sip and Qt modules

This moves the `import sip` statements in uhd_fft and uhd_siggen_gui
below the import for Qt, without which import errors can occur on
platforms such as Fedora 31.
---
 gr-uhd/apps/uhd_fft        | 3 +--
 gr-uhd/apps/uhd_siggen_gui | 5 +----
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/gr-uhd/apps/uhd_fft b/gr-uhd/apps/uhd_fft
index c9607670cf..4c2516c8ce 100755
--- a/gr-uhd/apps/uhd_fft
+++ b/gr-uhd/apps/uhd_fft
@@ -38,11 +38,10 @@ from __future__ import print_function
 from __future__ import division
 import ctypes
 import sys
-import sip
 import threading
 import time
-from distutils.version import StrictVersion
 from PyQt5 import Qt
+import sip # Needs to be imported after PyQt5, could fail otherwise
 from gnuradio import eng_notation
 from gnuradio import eng_arg
 from gnuradio import gr
diff --git a/gr-uhd/apps/uhd_siggen_gui b/gr-uhd/apps/uhd_siggen_gui
index 48a14df81a..fd4b781759 100755
--- a/gr-uhd/apps/uhd_siggen_gui
+++ b/gr-uhd/apps/uhd_siggen_gui
@@ -33,16 +33,14 @@ Signal Generator App
 ##################################################
 
 from __future__ import print_function
-import sip
 import sys
 import threading
 import time
-from distutils.version import StrictVersion
 from PyQt5 import Qt
 from PyQt5.QtCore import pyqtSlot
+import sip # Needs to be imported after PyQt5, could fail otherwise
 from gnuradio import analog
 from gnuradio import eng_notation
-from gnuradio import gr
 from gnuradio import qtgui
 from gnuradio import uhd
 from gnuradio.filter import firdes
@@ -473,4 +471,3 @@ def x11_init_threads():
 if __name__ == '__main__':
     x11_init_threads()
     main()
-
-- 
2.20.1

